mlflow/store/tracking/sqlalchemy_store.py

```diff
@@ -552,8 +552,10 @@ def update_run_info(self, run_id, run_status, end_time, run_name):
         with self.ManagedSessionMaker() as session:
             run = self._get_run(run_uuid=run_id, session=session)
             self._check_run_is_active(run)
-            run.status = RunStatus.to_string(run_status)
-            run.end_time = end_time
+            if run_status is not None:
```

Comment:
Does file store handle this correctly?

Reply:
Yes. File store backend invokes https://github.com/mlflow/mlflow/blob/de34d7844a08467ef0339f5441b186fae4adfb78/mlflow/entities/run_info.py#L76 that handle it perfectly.